runtime/vm/jvminit.c

```diff
@@ -2360,8 +2362,16 @@ static UDATA checkArgsConsumed(J9PortLibrary* portLibrary, J9VMInitArgs* j9vm_ar
 		}
 	}
 
+	xxIgnoreUnrecognizedXXOptionsDisableIndex = findArgInVMArgs( PORTLIB, j9vm_args, EXACT_MATCH, VMOPT_XXIGNOREUNRECOGNIZEDXXOPTIONSDISABLE, NULL, TRUE);
+	if (xxIgnoreUnrecognizedXXOptionsDisableIndex >= 0) {
+		xxIgnoreUnrecognizedXXOptionsEnableIndex = findArgInVMArgs( PORTLIB, j9vm_args, EXACT_MATCH, VMOPT_XXIGNOREUNRECOGNIZEDXXOPTIONSENABLE, NULL, TRUE);
+		if (xxIgnoreUnrecognizedXXOptionsDisableIndex > xxIgnoreUnrecognizedXXOptionsEnableIndex) {
+			ignoreUnrecongizedXXOption = JNI_FALSE;
+		}
+	}
+
 	for (i=0; i<j9vm_args->nOptions; i++) {
-		if (IS_CONSUMABLE( j9vm_args, i ) && !IS_CONSUMED( j9vm_args, i )) {
+		if (IS_IGNORED(j9vm_args,i) || (IS_CONSUMABLE( j9vm_args, i ) && !IS_CONSUMED( j9vm_args, i ))) {	
```

Comment:
Shouldn't this be `if (!ignored || (consumable && !consumed))`?  ie: only process args that haven't been ignored or already consumed?

Reply:
ah, that is correct, I will address this is my next push

Reply:
Looking back at this, I think the proper boolean statement is 

`!IS_IGNORED(j9vm_args,i) && IS_CONSUMABLE( j9vm_args, i ) && !IS_CONSUMED( j9vm_args, i )`

Reply:
I've tested the new boolean statement, and it seems to work as intended. I'll confirm with an internal grinder instance

Reply:
I don't think this code is necessary. As I understand it, the problem is that some options are parsed without being consumed. Once this problem is fixed, I don't expect the IGNORED flag will be needed. Until the problem is fixed, the IGNORED flag as written isn't going to make it work properly for all cases.

Reply:
Okay, I will be removing the IGNORED flag since it is no longer needed.